### PR TITLE
Add new parameters to .ini file to read them by SDL

### PR DESCRIPTION
**Pull request by dcherniev**
Added new parameters 'menuIcon' and 'menuTitle' to the .ini file.
Added to SDL ability to read default values for those parameters from ini file.
Also added to SDL resetting of those parameters in ResetGlobalProperties request.

Related tasks: [APPLINK-28166](https://adc.luxoft.com/jira/browse/APPLINK-28166), [APPLINK-28169](https://adc.luxoft.com/jira/browse/APPLINK-28169)

### CODE REVIEW

**Comment by dcherniev** [comment]
Please review : @vlantonov, @Kozoriz, @LuxoftAKutsan, @anosach-luxoft, @dtrunov, @VVeremjova 
Please note: I am currently working on remaining coverage by unit tests.

**Comment by dtrunov** [comment]
@dcherniev In describtion of PR please use [Appilnk-19101](link to issue), because I can not go to this task from your PR.

File: src/appMain/smartDeviceLink.ini [hunk 1]
```diff
@@ -211,6 +211,10 @@ ApplicationListUpdateTimeout = 2000
 ThreadPoolSize = 1
 ; The max size of hash which is used by OnHashUpdated
 HashStringSize = 32
+; Menu title default text
+MenuTitle = MENU
+; Path to default menu icon
+MenuIcon =
```
Comment (dcherniev) [inline src/appMain/smartDeviceLink.ini]: @dtrunov This is a requirement of the CRQ: 
MenuIcon default value should be empty (APPLINK-22707)

**Comment by dcherniev** [comment]
@dtrunov Please recheck the example link (Appilnk-19101), it doesn't work for me.

**Comment by dcherniev** [comment]
Unit tests are complete, please review : @vlantonov, @Kozoriz, @LuxoftAKutsan, @anosach-luxoft, @dtrunov, @VVeremjova

File: src/components/config_profile/test/profile_test.cc [hunk 2]
```diff
@@ -201,10 +201,12 @@ TEST_F(ProfileTest, UpdateBoolValues) {
...
   // Default values
   std::string recording_file_name = "record.wav";
+  std::string server_address = "127.0.0.1";
+  std::string app_resource_folder = "";
   std::string tts_delimiter_ = "";
   std::string vr_help_title_ = "";
-  std::string app_resource_folder = "";
-  std::string server_address = "127.0.0.1";
+  std::string menu_title = "";
+  std::string menu_icon = "";
```
Comment (vlantonov) [inline src/components/config_profile/test/profile_test.cc]: @dcherniev By default `std::string` is constructed empty, so there is no need to assign `""`

File: src/appMain/smartDeviceLink.ini (hunk 1 above)
Comment (dtrunov) [inline src/appMain/smartDeviceLink.ini]: @dcherniev ok

File: src/components/application_manager/src/commands/mobile/reset_global_properties_request.cc [hunk 3]
```diff
@@ -131,13 +131,18 @@ void ResetGlobalPropertiesRequest::Run() {
...
     }
-    // TODO(DT): clarify the sending parameter menuIcon
-    // if (menu_icon) {
-    //}
+    if (menu_icon) {
+      smart_objects::SmartObject so_menu_icon(smart_objects::SmartType_Map);
+      so_menu_icon[strings::value] =
+          application_manager_.get_settings().menu_icon();
+      msg_params[hmi_request::menu_icon] = so_menu_icon;
+      app->set_menu_icon(msg_params[hmi_request::menu_icon]);
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/commands/mobile/reset_global_properties_request.cc]: @dcherniev why you use nested menu icon if you can use so_menu_icon variable?

File: src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc [hunk 4]
```diff
@@ -77,6 +77,9 @@ typedef SharedPtr<ResetGlobalPropertiesResponse>
 namespace {
 const uint32_t kConnectionKey = 2u;
 const uint32_t kCorrelationId = 10u;
+const std::string kTimeOutDefaultValue = "time_out";
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc]: @dcherniev please rename it to kTimeOutPromtDefaultValue
Because it is confusing that kTimeOutDefaultValue is string

File: src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc [hunk 5]
```diff
@@ -198,59 +204,82 @@ TEST_F(ResetGlobalPropertiesRequestTest, Run_SUCCESS) {
...
+  // mobile_apis::GlobalProperty::MENUICON:
+  smart_objects::SmartObject so_menu_icon =
+      smart_objects::SmartObject(smart_objects::SmartType_Map);
+  EXPECT_CALL(mock_app_manager_settings_, menu_icon())
+      .WillOnce(ReturnRef(kMenuIconDefaultValue));
+  so_menu_icon[am::strings::value] = kMenuIconDefaultValue;
+  ui_msg_params[am::hmi_request::menu_icon] = so_menu_icon;
   EXPECT_CALL(*mock_app_,
-              set_menu_title(msg_params[am::hmi_request::menu_title]));
+              set_menu_icon(ui_msg_params[am::hmi_request::menu_icon]));
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc]: @dcherniev again. why you use nested variable if you can use existing?

File: src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc [hunk 6]
```diff
@@ -198,59 +204,82 @@ TEST_F(ResetGlobalPropertiesRequestTest, Run_SUCCESS) {
...
+      .WillOnce(ReturnRef(kMenuTitleDefaultValue));
+  ui_msg_params[am::hmi_request::menu_title] = kMenuTitleDefaultValue;
+  EXPECT_CALL(*mock_app_,
+              set_menu_title(ui_msg_params[am::hmi_request::menu_title]));
 
+  // mobile_apis::GlobalProperty::MENUICON:
+  smart_objects::SmartObject so_menu_icon =
+      smart_objects::SmartObject(smart_objects::SmartType_Map);
+  EXPECT_CALL(mock_app_manager_settings_, menu_icon())
+      .WillOnce(ReturnRef(kMenuIconDefaultValue));
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc]: @dcherniev please rename this values from Default to Test variables.

File: src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc [hunk 7]
```diff
@@ -308,7 +337,7 @@ TEST_F(ResetGlobalPropertiesRequestTest,
   (*msg_)[am::strings::msg_params][am::strings::properties][0] =
       mobile_apis::GlobalProperty::TIMEOUTPROMPT;
   (*msg_)[am::strings::msg_params][am::strings::properties][1] =
-      mobile_apis::GlobalProperty::MENUICON;
+      mobile_apis::GlobalProperty::MENUNAME;
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc]: @dcherniev in couple of files you changes all menu_name to menu icon, there  you do conversely. Is it error?

File: src/components/application_manager/src/commands/mobile/reset_global_properties_request.cc [hunk 8]
```diff
@@ -131,13 +131,18 @@ void ResetGlobalPropertiesRequest::Run() {
...
+          application_manager_.get_settings().menu_title();
       app->set_menu_title(msg_params[hmi_request::menu_title]);
     }
-    // TODO(DT): clarify the sending parameter menuIcon
-    // if (menu_icon) {
-    //}
+    if (menu_icon) {
+      smart_objects::SmartObject so_menu_icon(smart_objects::SmartType_Map);
+      so_menu_icon[strings::value] =
+          application_manager_.get_settings().menu_icon();
```
Comment (dtrunov) [inline src/components/application_manager/src/commands/mobile/reset_global_properties_request.cc]: @dcherniev Please read comments above

File: src/components/config_profile/test/smartDeviceLink.ini [hunk 9]
```diff
@@ -168,6 +168,10 @@ FrequencyTime = 1000
 ApplicationListUpdateTimeout = 2
 ; Max allowed threads for handling mobile requests. Currently max allowed is 2
 ThreadPoolSize = 1
+; Menu title default text
+MenuTitle = MENU
```
Comment (dtrunov) [inline src/components/config_profile/test/smartDeviceLink.ini]: @dcherniev Why did you write this line in ApplicationManager section. I think section GLOBAL PROPERTIES is more suitable.

File: src/components/config_profile/include/config_profile/profile.h [hunk 10]
```diff
@@ -581,6 +581,16 @@ class Profile : public protocol_handler::ProtocolHandlerSettings,
    */
   uint16_t open_attempt_timeout_ms_resumption_db() const OVERRIDE;
 
+  /**
+   * @brief Returns menu title
+   */
+  const std::string& menu_title() const;
```
Comment (dtrunov) [inline src/components/config_profile/include/config_profile/profile.h]: @dcherniev Please use OVERRIDE

File: src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc (hunk 7 above)
Comment (dcherniev) [inline src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc]: @LuxoftAKutsan Well this change here is not critical: presence of either MENUICON or MENUNAME was nesessary to send ui notification which was needed for specific case in on_event function, MENUICON realization used to be empty, so it was used here to create empty ui notification. I changed it to MENUNAME since it has slightly shorter realization (over MENUICON, which is now not empty).

File: src/components/config_profile/test/smartDeviceLink.ini (hunk 9 above)
Comment (dcherniev) [inline src/components/config_profile/test/smartDeviceLink.ini]: @dtrunov The [INI file][ApplicationManager] section was in requirements for both new fields ([APPLINK-22706](https://adc.luxoft.com/jira/browse/APPLINK-22706) and [APPLINK-22707](https://adc.luxoft.com/jira/browse/APPLINK-22707))

Comment (dtrunov) [inline src/components/config_profile/test/smartDeviceLink.ini]: @dcherniev Ok

File: src/components/application_manager/src/commands/mobile/reset_global_properties_request.cc [hunk 11]
```diff
@@ -131,13 +131,18 @@ void ResetGlobalPropertiesRequest::Run() {
       }
       msg_params = *vr_help;
     }
-    if (menu_name) {
-      msg_params[hmi_request::menu_title] = "";
+    if (menu_title) {
+      msg_params[hmi_request::menu_title] =
+          application_manager_.get_settings().menu_title();
       app->set_menu_title(msg_params[hmi_request::menu_title]);
```
Comment (dcherniev) [inline src/components/application_manager/src/commands/mobile/reset_global_properties_request.cc]: @dtrunov Clarifyed: it is expected by Ford that all values in .ini file shall be valid, and validation of .ini file is out of scope of the CRQ. So there should be no problems in reading or resetting of this parameter.

File: src/components/application_manager/src/commands/mobile/reset_global_properties_request.cc (hunk 8 above)
Comment (dcherniev) [inline src/components/application_manager/src/commands/mobile/reset_global_properties_request.cc]: @dtrunov Maybe you meant: "Please read comments below"?

**Comment by dcherniev** [comment]
Closing pull request due to moving of this feature into develop branch.

File: src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc (hunk 4 above)
Comment (dcherniev) [inline src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc]: @LuxoftAKutsan Fixed, please check 22b055f

File: src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc (hunk 6 above)
Comment (dcherniev) [inline src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc]: @LuxoftAKutsan Fixed, please check 22b055f

File: src/components/config_profile/test/profile_test.cc (hunk 2 above)
Comment (dcherniev) [inline src/components/config_profile/test/profile_test.cc]: @vlantonov Fixed, please check 22b055f

File: src/components/application_manager/src/commands/mobile/reset_global_properties_request.cc (hunk 8 above)
Comment (dcherniev) [inline src/components/application_manager/src/commands/mobile/reset_global_properties_request.cc]: @dtrunov Please clarify: what comments have you ment here?

File: src/components/config_profile/test/profile_test.cc (hunk 2 above)
Comment (vlantonov) [inline src/components/config_profile/test/profile_test.cc]: @dcherniev This does not seem to be fixed.

**Comment by dcherniev** [comment]
Please note: I have created a new pull request (after moving of this feature into develop branch). So requested changes should be present in pull request #1038
